test-integration/operator-tests/src/test/java/org/infinispan/operator/CustomLibsIT.java

```diff
@@ -0,0 +1,107 @@
+import cz.xtf.core.openshift.OpenShiftWaiters;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.*;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.assertj.core.api.Assertions;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Collections;
+
+@Slf4j
+@CleanBeforeAll
+public class CustomLibsIT {
+    private static final OpenShift openShift = OpenShifts.master();
+    private static final OpenShiftWaiters openShiftWaiters = OpenShiftWaiters.get(openShift, () -> false);
+    private static final Infinispan infinispan = Infinispans.customLibs();
+
+    @BeforeAll
+    static void prepare() throws IOException {
+        preparePVC();
+        preparePod();
+
+        openShiftWaiters.areExactlyNPodsRunning(1).waitFor();
+
+        buildLibs();
+        uploadLibs();
+
+        infinispan.deploy();
+        infinispan.waitFor();
+    }
+
+    @Test
+    void test() {
+        Pod pod = openShift.getPod("custom-libs-0");
+        String logs = openShift.getPodLog(pod);
+
+        Assertions.assertThat(logs).contains("Loaded extension 'basic-filter-factoy'");
+    }
+
+    private static void preparePVC() {
+        PersistentVolumeClaimBuilder pvc = new PersistentVolumeClaimBuilder();
+        pvc.withNewMetadata().withName("infinispan-libs").endMetadata();
+        pvc.withNewSpec()
+                .withAccessModes("ReadWriteOnce")
+                .withNewResources().withRequests(Collections.singletonMap("storage", new Quantity("100Mi")))
+                .endResources().endSpec();
+
+        openShift.createPersistentVolumeClaim(pvc.build());
+    }
+
+    private static void preparePod() {
+        String serverImage = XTFConfig.get("org.infinispan.image.server", "quay.io/infinispan/server:latest");
+
+        VolumeBuilder vb = new VolumeBuilder();
+        vb.withName("lib-pv-storage").withNewPersistentVolumeClaim("infinispan-libs", false);
+
+        VolumeMountBuilder vmb = new VolumeMountBuilder();
+        vmb.withMountPath("/tmp/libs");
+        vmb.withName("lib-pv-storage");
+
+        ContainerBuilder cb = new ContainerBuilder();
+        cb.withName("lib-pv-container")
+                .withImage(serverImage)
+                .withVolumeMounts(vmb.build());
+
+        PodBuilder pod = new PodBuilder();
+        pod.withNewMetadata().withName("infinispan-libs").endMetadata();
+        pod.withNewSpec()
+                .withNewSecurityContext().withFsGroup(2000L).endSecurityContext()
+                .withVolumes(vb.build())
+                .withContainers(cb.build())
+                .endSpec();
+
+        openShift.createPod(pod.build());
```

Comment:
Also test (probably others) didn't cleanup resources in case of fails. It would be nice to have this option.

Reply:
Each test contains `@CleanBeforeAll` which is responsible for cleaning up namespace resources before anything in the test is called. But it's true that tests do no cleanup after themselves beside removing Inifnispan CR to validate that everything was removed properly.

Shouldn't be an issue to add similar optional functionality for after the execution

Reply:
`@CleanBeforeAll` seem to very danger method which can be cleanup all resources in namespace

Reply:
That's intentional design to keep namespaces clean for running the tests and the tests only. It should also prevent any unintentional leftovers by any other tests in the namespace

Reply:
Yes, true, but limits to have a separate namespace for operator deployment. Not able to test in simple way. This design could  have but a bit limiting with short tests.

Reply:
I agree. I would actually like to make the Operator installation part of the test execution altogether with different install modes testing but that's out of scope for now as there are higher priority tasks